### PR TITLE
JUP-168 new withdrawal endpoint

### PR DESCRIPTION
**Pull request by arijoon**
Add the V4 withdrawal flow, for a lack of better name, I've named all functions as `transferAndWithdraw` to be clear it has to sign a transfer TX

### CODE REVIEW

**Review by Matko95 — APPROVED** [review]

File: src/api/transferAndWithdrawUsingVaultId.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+const { post } = require('request-promise')
+
+module.exports = async (dvf, transferData) => {
+  const url = dvf.config.api + '/v2/trading/w/transferAndWithdraw'
```
Comment (Neozaru) [inline src/api/transferAndWithdrawUsingVaultId.js:4]: Probably `/v1/trading/w/...` ?

**Review by Neozaru — APPROVED** [review]
Looks OK to me
Note that you can reference commit hash of this PR from dvf-portal and test there before even merging this. Usually helpful to avoid unexpected issues that require fixes on dvf-client-js

**Comment by arijoon** [comment]
> Looks OK to me Note that you can reference commit hash of this PR from dvf-portal and test there before even merging this. Usually helpful to avoid unexpected issues that require fixes on dvf-client-js

Indeed I have already tested locally, just wanted the verified hash to use before merging

File: src/api/transferAndWithdrawUsingVaultId.js (hunk 1 above)
Comment (arijoon) [reply src/api/transferAndWithdrawUsingVaultId.js:4]: hmm, this worked in local UI, but it does seem like it should be v1/trading

Comment (arijoon) [reply src/api/transferAndWithdrawUsingVaultId.js]: that was actually an unused file, so removed. nice catch

**Pull request merged**